l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -0,0 +1,309 @@
+                contractL2: _request.l2Contract,
+                mintValue: _request.mintValue,
+                l2Value: _request.l2Value,
+                l2Calldata: _request.l2Calldata,
+                l2GasLimit: _request.l2GasLimit,
+                l2GasPerPubdataByteLimit: _request.l2GasPerPubdataByteLimit,
+                factoryDeps: _request.factoryDeps,
+                refundRecipient: refundRecipient
+            })
+        );
+    }
+
+    /// @notice After depositing funds to the sharedBridge, the secondBridge is called
+    ///  to return the actual L2 message which is sent to the Mailbox.
+    ///  This assumes that either ether is the base token or
+    ///  the msg.sender has approved the sharedBridge with the mintValue,
+    ///  and also the necessary approvals are given for the second bridge.
+    /// @notice The logic of this bridge is to allow easy depositing for bridges.
+    /// Each contract that handles the users ERC20 tokens needs approvals from the user, this contract allows
+    /// the user to approve for each token only its respective bridge
+    /// @notice This function is great for contract calls to L2, the secondBridge can be any contract.
+    function requestL2TransactionTwoBridges(
+        L2TransactionRequestTwoBridgesOuter calldata _request
+    ) external payable override nonReentrant returns (bytes32 canonicalTxHash) {
+        {
+            address token = baseToken[_request.chainId];
+            uint256 baseTokenMsgValue;
+            if (token == ETH_TOKEN_ADDRESS) {
+                require(msg.value == _request.mintValue + _request.secondBridgeValue, "Bridgehub: msg.value mismatch");
+                // kl todo it would be nice here to be able to deposit weth instead of eth
+                baseTokenMsgValue = _request.mintValue;
+            } else {
+                require(msg.value == _request.secondBridgeValue, "Bridgehub: msg.value mismatch 2");
+                baseTokenMsgValue = 0;
+            }
+            sharedBridge.bridgehubDepositBaseToken{value: baseTokenMsgValue}(
+                _request.chainId,
+                msg.sender,
+                token,
+                _request.mintValue
+            );
+        }
+
+        address stateTransition = getStateTransition(_request.chainId);
+
+        L2TransactionRequestTwoBridgesInner memory outputRequest = IL1SharedBridge(_request.secondBridgeAddress)
+            .bridgehubDeposit{value: _request.secondBridgeValue}(
+            _request.chainId,
+            msg.sender,
+            _request.l2Value,
+            _request.secondBridgeCalldata
+        );
+
+        require(outputRequest.magicValue == TWO_BRIDGES_MAGIC_VALUE, "Bridgehub: magic value mismatch");
+
+        address refundRecipient = _actualRefundRecipient(_request.refundRecipient);
+
+        require(_request.secondBridgeAddress > address(1000), "Bridgehub: second bridge address too low"); // to avoid calls to precompiles
```

Comment:
Here better to introduce constant instead of using `address(1000)`. Also I would use `type(uint16).max` instead as it is consistent with [eip-1352](https://eips.ethereum.org/EIPS/eip-1352) that is implemented in L2